### ISSUE TITLE
False positive detection of an unused variable

### ISSUE THREAD

**Issue by Yura52** [astral-sh/ty#3193]
### Summary

In the following snippet, in the line `x = i`, ty marks `x` as unused.

```python
def f() -> None:
    x = None
    for i in range(10):
        if x is None:
            x = i  # <-- ty says that `x` is unused
        else:
            print(x)
```

Software versions:

- Python: 3.13.12
- ty: 0.0.27

### Version

0.0.27

**Comment by charliermarsh** [comment]
I believe this is fixed by https://github.com/astral-sh/ruff/pull/24336.
